### PR TITLE
Improve module docs for error handler middleware

### PR DESCRIPTION
**Pull request by mlodato517**
**This Commit**
Updates the docs in the error handler middleware to match
[changes made to the website][website].

**Why?**
The previous documentation, when run, doesn't give the user a trivial
way to exercise the middleware because none of the routes return a
status code covered by the error handler middleware. There are also
additional routes that don't seem obviously instructive and so may
confuse a user into thinking error handlers only work with certain kinds
of requests.

[website]: https://github.com/actix/actix-website/pull/248

## PR Type

Documentation

## PR Checklist

- [x] Documentation comments have been added / updated.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

### CODE REVIEW

File: src/middleware/err_handlers.rs
Comment (mlodato517) [inline src/middleware/err_handlers.rs:40]: I reordered these imports according to `rustfmt`.

File: src/middleware/err_handlers.rs
Comment (mlodato517) [inline src/middleware/err_handlers.rs:44]: I renamed this function - see https://github.com/actix/actix-website/pull/248 for more details.

File: src/middleware/err_handlers.rs
Comment (mlodato517) [inline src/middleware/err_handlers.rs:47]: I copied this into a non-rustdoc setting and let `rustfmt` reformat it to what you see below.

File: src/middleware/err_handlers.rs
Comment (mlodato517) [inline src/middleware/err_handlers.rs:53]: This reformatting was also performed by `cargo fmt` by copy-pasting into a fresh repo. I'm not sure if we keep doc comments here tighter than 100 characters so I'll happily unformat if we prefer that!

**Comment by mlodato517** [comment]
I wasn't sure if I needed a changelog entry for this but I'm happy to do that. I was also unsure how to add labels ... I don't obviously see an active button in the UI :thinking:

**Comment by mlodato517** [comment]
I was waiting to un-draft this until CI passed but it looks like the actions won't run until a maintainer approves :sweat_smile: Which is all well and good - it'd be nice to not run all the benchmarks and things for this doc-only change anyway haha.

**Review by robjtede — APPROVED** [review]
thanks :)

**Pull request merged**
